llm4pddl/envs/assets/pddl/dressed/generate_tasks.py

```diff
@@ -0,0 +1,199 @@
+                               num_formal_in_suit: int, num_extra_objects: int,
+                               rng: np.random.Generator) -> str:
+    """Generates a single problem with the given parmeters."""
+    assert num_people >= (num_casual_events + num_formal_in_dress +
+                          num_formal_in_suit)
+    # deciding extra clothing:
+    extras = ['ex_p', 'ex_d', 'ex_sp', 'ex_ss', 'ex_np', 'ex_cs', 'ex_sj']
+    exs = {ex: 0 for ex in extras}
+    for _ in range(num_extra_objects):
+        exs[rng.choice(extras)] += 1
+    # making objects:
+    persons = [f'person{i}' for i in range(1, num_people + 1 + exs['ex_p'])]
+    dresses = [
+        f'dress{i}' for i in range(1, num_formal_in_dress + 1 + exs['ex_d'])
+    ]
+    sweatpants = [
+        f'sweatpants{i}'
+        for i in range(1, num_casual_events + 1 + exs['ex_sp'])
+    ]
+    sweatshirts = [
+        f'sweatshirt{i}'
+        for i in range(1, num_casual_events + 1 + exs['ex_ss'])
+    ]
+    nicepants = [
+        f'nice-pants{i}'
+        for i in range(1, num_formal_in_suit + 1 + exs['ex_np'])
+    ]
+    collaredshirt = [
+        f'collared-shirt{i}'
+        for i in range(1, num_formal_in_suit + 1 + exs['ex_cs'])
+    ]
+    suitjacket = [
+        f'suit-jacket{i}'
+        for i in range(1, num_formal_in_suit + 1 + exs['ex_d'])
+    ]
+    objects = [{
+        'type': 'person',
+        'objs': persons
+    }, {
+        'type': 'dress',
+        'objs': dresses
+    }, {
+        'type': 'sweatpants',
+        'objs': sweatpants
+    }, {
+        'type': 'sweatshirt',
+        'objs': sweatshirts
+    }, {
+        'type': 'nice-pants',
+        'objs': nicepants
+    }, {
+        'type': 'collared-shirt',
+        'objs': collaredshirt
+    }, {
+        'type': 'suit-jacket',
+        'objs': suitjacket
+    }]
+    object_str = ''
+    for obj in objects:
+        if len(obj['objs']) > 0:
+            object_str += ' '.join(
+                obj['objs']) + f' - {obj["type"]}\n            '
+    # making initial predicates:
+    init_preds = ''
+    for person in persons:
+        init_preds += (f'(wearing-nothing-formal {person})\n         ' +
+                       f'(wearing-nothing-casual {person})\n         ')
+    for clothing in (dresses + sweatpants + sweatshirts + nicepants +
+                     collaredshirt + suitjacket):
+        init_preds += f'(in-closet {clothing})' + '\n         '
+    # making goal predicates:
+    goals = ''
+    goal_predicates = []
+    rng.shuffle(persons)
+    for i in range(num_casual_events):
+        goal_predicates.append(
+            f'(attending-casual-event {persons[i]})\n              ')
+    for i in range(
+            num_casual_events,
+        (num_casual_events + num_formal_in_dress + num_formal_in_suit)):
+        goal_predicates.append(
+            f'(attending-formal-event {persons[i]})\n              ')
+    rng.shuffle(goal_predicates)
```

Comment:
why shuffle here?

Reply:
My thought process is that I want the goal ground atoms to be given in a randomized order and, in this case, to randomized people so that across the problems there is less structure(for example person1 always attending a casual event, etc) to who gets what events. What do you think about this?

Reply:
Randomizing the people / event makes sense, so the shuffle you have before this one is good. But I think this is just going to shuffle the order that the ground atoms appear in the goal? And since the goal is an unordered set, that's just gonna be a superficial difference. So shuffling is fine, but I would expect it to have no effect.

Reply:
Yeah I definitely agree that it wont impact any of the traditional planners from solving...I mostly did this so that this string will be scrambled some more when it gets inputted to GPT-3 just in case it somehow latches onto a representation that has casual events being before formal events in the goal portion of the string. But yeah, this shouldn't change anything at all.

Reply:
That's a good point! I hadn't thought of that. Worth keeping it then.